src/enforcers/LogicalOrWrapperEnforcer.sol

```diff
@@ -0,0 +1,243 @@
+        bytes calldata _args,
+        ModeCode _mode,
+        bytes calldata _executionCallData,
+        bytes32 _delegationHash,
+        address _delegator,
+        address _redeemer
+    )
+        public
+        override
+        onlyDefaultExecutionMode(_mode)
+    {
+        _executeHook(
+            _terms,
+            _args,
+            Params(_mode, _executionCallData, _delegationHash, _delegator, _redeemer),
+            CaveatEnforcer.afterHook.selector
+        );
+    }
+
+    /**
+     * @notice Hook called after all delegations are executed
+     * @dev Validates that the execution mode is default and calls the appropriate hook on all caveats
+     * @param _terms Encoded array of CaveatGroup, where each group contains multiple caveats to evaluate
+     * @param _args Encoded SelectedGroup specifying which group to evaluate and arguments for its caveats
+     * @param _mode The execution mode
+     * @param _executionCallData The execution data
+     * @param _delegationHash The hash identifying the delegation
+     * @param _delegator The address of the delegator
+     * @param _redeemer The address redeeming the delegation
+     */
+    function afterAllHook(
+        bytes calldata _terms,
+        bytes calldata _args,
+        ModeCode _mode,
+        bytes calldata _executionCallData,
+        bytes32 _delegationHash,
+        address _delegator,
+        address _redeemer
+    )
+        public
+        override
+        onlyDefaultExecutionMode(_mode)
+    {
+        _executeHook(
+            _terms,
+            _args,
+            Params(_mode, _executionCallData, _delegationHash, _delegator, _redeemer),
+            CaveatEnforcer.afterAllHook.selector
+        );
+    }
+
+    ////////////////////////////// Internal Methods //////////////////////////////
+
+    /**
+     * @notice Internal function that executes the specified hook on all caveats in a group
+     * @dev This function handles the core logic of decoding terms and args, validating indices,
+     *      and executing the specified hook on each caveat in the selected group
+     * @param _terms Encoded array of CaveatGroup, where each group contains multiple caveats to evaluate
+     * @param _args Encoded SelectedGroup specifying which group to evaluate and arguments for its caveats
+     * @param _params The consolidated parameters for the hook execution
+     * @param _hookSelector The function selector for the hook to execute
+     */
+    function _executeHook(bytes calldata _terms, bytes calldata _args, Params memory _params, bytes4 _hookSelector) internal {
+        CaveatGroup[] memory caveatGroups_ = abi.decode(_terms, (CaveatGroup[]));
+        SelectedGroup memory selectedGroup_ = abi.decode(_args, (SelectedGroup));
+
+        require(selectedGroup_.groupIndex < caveatGroups_.length, "LogicalOrWrapperEnforcer:invalid-group-index");
+
+        CaveatGroup memory caveatGroup_ = caveatGroups_[selectedGroup_.groupIndex];
```

Comment:
we should have an option here where the redeemer doesnt have to pass in a selected group. Perhaps if the selected group is empty then we loop through all caveat groups and execute them?